callhome.go

```diff
@@ -0,0 +1,157 @@
+package netconf
+
+import (
+	"crypto/tls"
+	"fmt"
+	"github.com/nemith/netconf/transport"
+	ncssh "github.com/nemith/netconf/transport/ssh"
+	nctls "github.com/nemith/netconf/transport/tls"
+	"golang.org/x/crypto/ssh"
+	"net"
+)
+
+type TransportType uint
+
+const (
+	TransportTypeSSH TransportType = iota
+	TransportTypeTLS
+)
+
+/*
+CallHome implements netconf callhome procedure as specified in RFC 8071
+*/
+type CallHome struct {
+	listener  net.Listener
+	network   string
+	addr      string
+	transport TransportType
+	configSSH *ssh.ClientConfig
+	configTLS *tls.Config
+	session   *Session
```

Comment:
One thing about a callhome server is that it should support multiple clients.  This should be a map (protected with a mutex probably) and the `CallHomeServer` object should have ways of obtaining a particular client based on some sort of identifier.

Looking at the RFC indentifiers could be the the certificate FP (tls) or host key (ssh) and or ipaddress/port pair?